src/app/common/components/MustGatherModal.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import { Button, TextContent, Modal } from '@patternfly/react-core';
+import { MustGatherContext, AppContext, mustGatherObjType } from '@app/common/context';
+import { useMustGatherMutation } from '@app/queries';
+
+export const MustGatherModal: React.FunctionComponent = () => {
+  const mustGatherContext = React.useContext(MustGatherContext);
+  const { pushNotification } = React.useContext(AppContext);
+
+  const handleMustGatherSuccess = () => {
+    if (mustGatherContext.activeMustGather) {
+      pushNotification({
```

Comment:
Just thinking out loud, because I like this: in other places where we have a modal that calls a mutation, we show the result of the mutation by keeping the modal open until the mutation resolves succesfully, and showing any errors as inline alerts in the modal (by using `<ResolvedQuery />`... incidentally we may want to factor out its error-parsing bits for use in push notifications).

Here though, the modal closes immediately and the result is shown as a push notification. I think it works here because you're kicking off a long-running process, and because there is no input in the modal. But it's inconsistent with our other mutation-modals... I'm trying to determine if we should open an issue to make them consistent later or if they should be different. I like the push notifications a lot and was about to suggest we switch to them everywhere, but in the other places I'm referring to, we want to preserve the form data in the modal so they can make a correction and try again if the mutation fails. 🤔 

It does make me think though that we should use a push notification for kicking off other long running tasks, like when a migration starts or is canceled, or maybe even when we detect a VM failing (yikes, that's a whole future project. Don't tell Miguel yet)

Reply:
ha it's a good point. Both strategies have a different UX and which one to use probably just depends on the use case. The notifications are a new feature, so I'm thinking not go too fast and put them everywhere all at once, but we should definitely make use of them anywhere it improves the flow with no downside.